src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -24,248 +20,314 @@
+        if (e.getCause() != null)
+            log(e.getCause(), level+1);
+    }
+
+    private void log(Throwable e)
+    {
+        log(e, 0);
+    }
+
+    public void setModel(IElevatorWrapper model) {
+        if (model == null)
+        {
+            return;
+        }
+        this.model = model;
+
+        //Init elevator floors
+        try {
+            info = model.queryGeneralInformation();
+        } catch (Exception e) {
+            log(new RuntimeException("Failed to query model!", e));
+            this.model = null;
+            return;
+        }
+
+        floorNames.clear();
+        gElevatorFloors.getChildren().clear();
+        floors = new FloorState[info.getNrOfFloors()];
+
+        lTopFloor.setText(((Integer) (info.getNrOfFloors() - 1)).toString());
+
+        for (int i = 0; i < info.getNrOfFloors(); i++) {
+            floorNames.add("Floor " + i);
+            floors[i] = new FloorState();
+
+            RowConstraints rCon = new RowConstraints();
+            rCon.setMinHeight(20);
+            rCon.setPercentHeight(100.0 / info.getNrOfFloors());
+
+            HBox hb = new HBox();
+            hb.setAlignment(Pos.CENTER_RIGHT);
+            hb.setSpacing(5);
+            hb.setPadding(new Insets(5, 5, 5, 5));
+            hb.translateXProperty().set(10);
+
+            Line line = new Line(0, 0, 10, 0);
+            line.disableProperty().bind(floors[i].isServiced.not());
+
+            Label label = new Label(Integer.toString(i));
+            label.disableProperty().bind(floors[i].isServiced.not());
+
+            hb.getChildren().addAll(
+                    createFloorImageView("images/arrowUp.png", floors[i].requestUp),
+                    createFloorImageView("images/arrowDown.png", floors[i].requestDown),
+                    createFloorImageView("images/hand.png", floors[i].stopRequest),
+                    createFloorImageView("images/filler.jpeg", targetFloor.isEqualTo(i)),
+                    label,
+                    line);
+
+            gElevatorFloors.getRowConstraints().add(rCon);
+            gElevatorFloors.add(hb, 0, info.getNrOfFloors() - i - 1);
+        }
+
+        elevators.clear();
+        for (int i = 0; i < info.getNrOfElevators(); i++) {
+            elevators.add("Elevator " + i);
+        }
+
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                update();
+            }
+        }, 0, 100);
+    }
+
+    private void update() {
+        if (model == null) {
+            log(new RuntimeException("Invalid model"));
+            return;
+        } else if (currentElevator.get() < 0)
+            return;
+
+        Platform.runLater(() -> {
+            ElevatorState elevatorState;
+
+            try {
+                elevatorState = model.queryElevatorState(currentElevator.intValue());
+            } catch (Exception e) {
+                log(new RuntimeException("Failed to query model!", e));
```

Comment:
Just log the exception itself. The model already localizes the error messages etc.

Reply:
done in 923198d7e5c451ca51cc84a4d4f0566f790c1a8c